### PR TITLE
Bitswap Complete (1.0.0 + 1.1.0)

### PR DESCRIPTION
**Pull request by diasdavid**
> Nothing to see here yet, just created to keep tabs and references in what needs to be done
- [ ] Needs to support both bitswap 1.1.0 and 1.0.0
- [x] Needs js-multistream to support custom validation functions (mainly semver)
- [ ] New Bitswap message.proto (supports both old and new)
  - https://github.com/ipfs/go-ipfs/blob/master/exchange/bitswap/message/pb/message.proto

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -304,3 +310,7 @@ module.exports = class Bitwap {
     this.engine.stop()
   }
 }
+
+module.exports = Bitswap
+
+function noop () {}
```
Comment (dignifiedquire) [inline src/index.js]: didn't we agree that we would stop doing style refactorings that are based on personal preferences?

File: src/message/index.js [hunk 2]
```diff
@@ -35,8 +35,9 @@ class BitswapMessage {
     }
   }
 
-  addBlock (block) {
-    this.blocks.set(mh.toB58String(block.key), block)
+  addBlock (block, hashAlg) {
+    const key = block.key(hashAlg)
```
Comment (dignifiedquire) [inline src/message/index.js]: this is going to be async with the new crypto. so might be better to base it on top of that

File: test/browser.js [hunk 3]
```diff
@@ -25,8 +25,8 @@ function createRepo (id, done) {
     })
   })
 
-  const mainBlob = new Store(id)
-  const blocksBlob = new Store(`${id}/blocks`)
+  const mainBlob = new Store(id + Math.random())
+  const blocksBlob = new Store(`${id}/blocks` + Math.random())
```
Comment (dignifiedquire) [inline test/browser.js]: why?

File: test/index-test.js [hunk 4]
```diff
@@ -46,7 +46,7 @@ module.exports = (repo) => {
     })
 
     describe('receive message', () => {
-      it('simple block message', (done) => {
+      it.only('simple block message', (done) => {
```
Comment (dignifiedquire) [inline test/index-test.js]: TODO: remove before merge

File: src/index.js (hunk 1 above)
Comment (diasdavid) [inline src/index.js]: I might be wrong, but I don't see this as a style preference. Seeing `cb = cb || noop` is more simpler than `cb = cb || (() => {})` everytime I want a noop.

I can revert back though.

As I said at the top of the PR: "Nothing to see here", which was meant to save time from CR, this is far from complete (note that I didn't add the "needs review" label).

**Comment by dignifiedquire** [comment]
> nothing to see here

...someone needs to pay better attention to headlines *cough Just ignore me till you are ready :)

File: test/browser.js (hunk 3 above)
Comment (diasdavid) [inline test/browser.js]: This was due to https://github.com/ipfs/js-ipfs/issues/528#issuecomment-255138513

**Pull request closed without merging**